### DISCUSSION TITLE
Store credit Payment Plugin

### DISCUSSION THREAD

**Discussion by dylviz** [vendurehq/vendure#2223] [Q&A]
Hi, just brainstorming an idea that I'll be needing to implement and hoping to get some quick feedback to point me in the right direction. 

So the scenario I'm trying to address is pretty simple and straight forward. A customer receives a digital code or key, then loads up their "store credit" onto the platform and can use the credit to make purchases. 

I could also go the route of using a gift card provider (I know Mollie offers gift card processing), but this looks like it cost moneys. 
So, I'm also thinking to create a new plugin for store-credit that basically implements one function `transfer(from, to, amount)` to handle some of the user's credit stored in the DB.

I can hardcode the keys or even issue the credit manually (on the DB) if needed since this is just for testing with about 10 people.
I wouldn't be surprised if this plugin already exists somewhere, but i haven't seen it and don't mind trying to create my first plugin either.

Any input or recommendation would be great. Mostly just want to confirm that it's a straight forward add-on like how I'm thinking it is. Thanks!

**Comment by michaelbromley** [comment]
Hi!

First of all, I need to let you know that there is a gift card / store credit plugin that we developed for our paid marketplace that will be released later this year, probably before Q3. It does the thing you are describing but also has other features you may not need - generation of personalized gift card artwork, automated emailing etc.

There is certainly scope for a community-driven open-source solution if you want to build something!

**Comment by dylviz** [comment]
Thank you for letting me know! I will take some more time to look into it to see if it's something I can make progress on but I'm not really familiar with Nestjs so don't hold your breath :smile:
